Make it clearer that extensions can only be installed during initial setup

Pabl0cks opened the pull request:
I've only changed the wording of the main section titles to "Starter extensions", because changing the childs titles may feel too repetitive. Also:

- Tried to enforce the idea of installation - initial project in the descriptions and subtitle.
- Added a visual hint in the main section and the `Available Extensions` section.

Closes #82

technophile-04 commented on docs/extensions/extensions.md:
```diff
@@ -2,23 +2,29 @@
-Extensions are modular add-ons for Scaffold-ETH 2 that provide additional functionality or serve as examples for specific features. They offer several benefits:
+Starter Extensions are modular add-ons for Scaffold-ETH 2 that provide additional functionality or serve as examples for specific features.
+
+:::info Hint
```
Same here as https://github.com/scaffold-eth/se-2-docs/pull/83/files#r1757072217 ?

Pabl0cks replied:
Updated! 🙌

technophile-04 reviewed (commented):
Tysm @Pabl0cks!! Umm not sure if "Starter Extension" seems a perfect word to me now I mean it makes sense just as word but for me it seems a bit weird while reading it in a sentence. 

Really love the idea of adding Info I think it it nicely puts out the point. 

I was thinking maybe for now just call them extension(as previously) and mention the Info? 

Maybe we can update the title/ name when we actually change the flag from `extension` to something else?  What do you think?

Pabl0cks commented on docs/extensions/availableExtensions.md:
```diff
@@ -4,6 +4,10 @@ sidebar_position: 10
 
 # Available Extensions
 
+:::info Hint
```
My initial thought was "Info" but then somehow I changed my mind and set "Hint" :)

Updated 🙏

Pabl0cks commented:
> Tysm @Pabl0cks!! Umm not sure if "Starter Extension" seems a perfect word to me now I mean it makes sense just as word but for me it seems a bit weird while reading it in a sentence.

Agree with that! It felt a bit strange when reading it again. Changed back to Extensions and updated Hint => Info 🙌

TYSM for the feeback ♥

technophile-04 commented:
Tysm @Pabl0cks!!
